src/hooks/useEnabledExtensions.tsx

```diff
@@ -1,56 +1,25 @@
-// import { Extension } from '@colony/colony-js';
-// import { useColonyExtensionsQuery } from '~data/index';
-// import { Address } from '~types/index';
+import { Extension } from '@colony/colony-js';
 
-// interface Props {
-//   colonyAddress?: Address;
-// }
+import useExtensionsData from './useExtensionsData';
 
-/*
- * @TODO This needs to be addressed and either refactored or removed
- */
+const useEnabledExtensions = () => {
+  const { installedExtensionsData, loading } = useExtensionsData();
 
-// export const useEnabledExtensions = ({ colonyAddress }: Props) => {
-//   const { data: colonyExtensionsData, loading } = useColonyExtensionsQuery({
-//     variables: { address: colonyAddress || '' },
-//   });
-//   const installedExtensions =
-//     colonyExtensionsData?.processedColony?.installedExtensions || [];
+  const oneTxPaymentData = installedExtensionsData.find(
+    (extensionData) => extensionData.extensionId === Extension.OneTxPayment,
+  );
+  const isOneTxPaymentEnabled = !!oneTxPaymentData?.isEnabled;
 
-//   const installedVotingExtension = installedExtensions.find(
-//     ({ extensionId }) => extensionId === Extension.VotingReputation,
-//   );
-//   const installedOneTxPaymentExtension = installedExtensions.find(
-//     ({ extensionId }) => extensionId === Extension.OneTxPayment,
-//   );
+  const votingReputationData = installedExtensionsData.find(
+    (extensionData) => extensionData.extensionId === Extension.VotingReputation,
+  );
+  const isVotingReputationEnabled = !!votingReputationData?.isEnabled;
 
-//   const isVotingExtensionEnabled = !!(
-//     installedVotingExtension &&
-//     installedVotingExtension.details?.initialized &&
-//     !installedVotingExtension.details?.deprecated
-//   );
-//   const isOneTxPaymentExtensionEnabled = !!(
-//     installedOneTxPaymentExtension &&
-//     installedOneTxPaymentExtension.details?.initialized &&
-//     !installedOneTxPaymentExtension.details?.deprecated
-//   );
-
-//   const installedExtensionsAddresses = installedExtensions.map((extension) =>
-//     extension.address?.toLowerCase(),
-//   );
-
-//   return {
-//     isVotingExtensionEnabled,
-//     votingExtensionVersion: installedVotingExtension
-//       ? installedVotingExtension?.details?.version
-//       : null,
-//     isOneTxPaymentExtensionEnabled,
-//     installedExtensionsAddresses,
-//     isLoadingExtensions: loading,
-//   };
-// };
-//
-
-const useEnabledExtensions = (...args) => args;
+  return {
```

Comment:
May I suggest this be generalised, so we needn't manually copy / paste the above each time we add a new extn?

E.g.
``` js
const enabledExtns = installedExtensionsData.reduce(
    (extns, extn) => ({
      ...extns,
      [`is${extn.extensionId}Enabled`]: extn.isEnabled,
    }),
    {},
  );

  return {
    loading,
    ...enabledExtns,
  };

Reply:
I'm just wondering if doing this won't cause more trouble down the line.
You're definitely right that we wouldn't have to modify this file when adding new supported extension but the way these are set up still requires doing quite a bit of copy pasting in other places.
My concern is that we'll lose strong typing, in your example if you try to access `enabledExtns.isOneTxPaymentEnabled` you will get the following type error: `Property 'isOneTxPaymentEnabled' does not exist on type '{}'` so I guess we would have to type it as `Record<string, boolean>` at best which I think isn't ideal. It also feels like we will be using this hook more often than we will be adding new extensions, so what are your thoughts on that?

Reply:
@jakubcolony How's this?

``` js
import { Extension } from '@colony/colony-js';

type Extensions = `${Extension}`;
type EnabledExtensions = Partial<Record<Extensions, boolean>>
```
then

``` js
     ...
      [`is${extn.extensionId}Enabled`]: extn.isEnabled,
    }),
    {} as EnabledExtensions,
  );
```
That's probably the most correct way to handle the typing?

Reply:
I like your approach, thanks for the suggestion